modules/reitit-core/src/reitit/impl.cljc

```diff
@@ -9,6 +9,51 @@
      (:import (java.net URLEncoder URLDecoder)
               (java.util HashMap Map))))
 
+;;
+;; path-update
+;;
+
+(defn -match [path path-map]
+  (letfn [(match [x f] (if (fn? f) (f x) (= x f)))]
+    (reduce
+     (fn [_ [ps f]]
+       (let [match (loop [[p & pr] path, [pp & ppr] ps]
+                     (cond (and p pp (match p pp)) (recur pr ppr)
+                           (= nil p pp) true))]
+         (when match (reduced f))))
+     nil path-map)))
+
+(defn -path-vals [m path-map]
+  (letfn [(-path-vals [l p m]
+            (reduce
+             (fn [l [k v]]
+               (let [p' (conj p k)
+                     f (-match p' path-map)]
+                 (cond
+                   f (cons [p' (f v)] l)
```

Comment:
Consider separating the map => assoc-in key-val pairs transform and the matching logic.

Reply:
considered, thanks.